### PR TITLE
Updated address component for edu benefits

### PR DESCRIPTION
**Pull request by jbalboni**
#2980.

This contains updates to the Address component for edu benefits. The form is updated and usable, but tests will fail until I make a schema update.

Changes are:
1. Country moved to top and defaulted to USA
2. States include military codes
3. Typing apo/fpo/dpo in city will narrow down state list (I think we're on the fence about this right now)
4. Single set of fields for whatever type of address
5. Postal code validation for US/Can (I'm not sure why we weren't doing this before, so we can take it out if I'm missing something).

### CODE REVIEW

**Comment by jbalboni** [comment]
@akainic @cweber @goldenmeanie Any feedback you have on this is welcome.

**Comment by akainic** [comment]
Looks great so far! I'm getting some weird errors when I play with this locally, though. There's something weird with the values of the fields. Sometimes it doesn't let me edit what I've entered, and other times it doesn't let me actually select something. Don't have time at this moment to dig into this more but can look more later! Just wanted to let you know.

**Comment by jbalboni** [comment]
@akainic @goldenmeanie Should be working better now.

**Comment by jbalboni** [comment]
Added validation to catch partial addresses

**Comment by cweber** [comment]
Looks pretty good to me. I couldn't trigger any errors in particular. Changing the country at top wipes out state still, but that makes sense.

Adding a space before or after dpo/apo gives you the full state list, but the options they need are right at the top anyway, so that seems ok.

**Comment by jbalboni** [comment]
@cweber Ah, good catch, I should trim the city before checking it. 

I intentionally reset state when the country changes, since it's unlikely to make sense anymore and prevents bad data from slipping through. We don't have to do that, though.

Though after thinking about it and trying it out, I don't think it's a big benefit to do the apo/fpo/dpo check.

**Comment by jbalboni** [comment]
@akainic This is ready for a full review now. CI is going to fail until I update the json schema, fyi.

**Comment by jbalboni** [comment]
@cweber @goldenmeanie @akainic How do you all feel about the apo/fpo/dpo city thing? I'm leaning towards removing it.

File: test/edu-benefits/components/Address.unit.spec.jsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  beforeEach(() => {
+    component = ReactTestUtils.renderIntoDocument(
+      <Address/>
+    );
+    assert.ok(component, 'Cannot even render component');
+  });
+
+  xit('has sane looking features', () => {
+    const inputs = ReactTestUtils.scryRenderedDOMComponentsWithTag(component, 'input');
+    expect(inputs).to.have.length(1);
```
Comment (akainic) [inline test/edu-benefits/components/Address.unit.spec.jsx]: I think there are more than 1 input in the `Address` component (I know this test is commented out anyway, can we re-comment it as well?)

**Review by akainic — COMMENTED** [review]
This looks great! 2 concerns:
1. When I test this locally, prefill doesn't seem to work for me (testing in Chrome). The prefill options show up as I type in the street address, but if I select one of them the fields aren't actually prefilled. Not sure why at first glance, the right attributes seem to be in place. Wondering if it's working for anyone else.
2. I'm a little worried about changing the schema for address fields to be different between apps (HCA still has `zipcode` for the ZIP field, but now edu will have `postalCode` for the ZIP field). I think in HCA we are required to differentiate between the 2. Just wondering if down the road this difference in field names between apps will create any problems? Like if we eventually have 1 veteran object from which we can prefill new forms? Maybe not something to worry about yet, but just something I was thinking about in looking at this.

**Comment by jbalboni** [comment]
@akainic I have that problem prefilling, but I thought it was something that was happening generally. That doesn't seem to be the case on vets.gov. I'll look into that a little more.

It sounded like from @markolson that HCA had specific requirements around the address that we didn't necessarily want to replicate. We definitely should try to avoid a proliferation of address components and schemas, but I'm hoping this one can be the standard, since it's the simplest.

File: test/edu-benefits/components/Address.unit.spec.jsx (hunk 1 above)
Comment (jbalboni) [inline test/edu-benefits/components/Address.unit.spec.jsx]: I completely missed that. It's updated now.

**Comment by jbalboni** [comment]
@akainic The autocomplete issue is baffling to me. It's only broken for edu-benefits. It was broken before this PR and I haven't had much luck tracking it down, so I think I'll open another issue for it.

**Comment by akainic** [comment]
@jbalboni works for me! Feel free to merge this one then.

**Pull request merged**